### ISSUE TITLE
TypeScript translation

### ISSUE THREAD

**Issue by davnat** [vedrxn/http-exception#1]
Hi,
I needed something like this library, but I need TS declarations.
So I ended up in translating all your files in TS (tests included, all passing): are you interested?
Cheers

**Comment by vdrn** [comment]
Hey, thanks for your interest. Yes, it's true the library doesn't have typescript declarations at version 0.0.2, sorry about that. I've been planning on migrating the project to typescript in the near future and adding declarations, but that's something I'd like to take on myself. Happy to hear your migrated tests are passing, be sure to look out for the http-exception typescript update.

**Comment by davnat** [comment]
I thought to add a note, then.
I started to use my translated library, and I've been a bit disappointed: the main point in using TS is having a working typechecking system, but to use my library as it is I'd have to bypass it, because it's adding methods to HttpException at runtime (well, I only spent a couple of hours on it, I cannot expect it to be any close to perfection). Not to talk about code completion.
I explored a bit the possibilities, and found [this](https://stackoverflow.com/a/51067065/12291793): it looks cool, but not a trivial implementation.
Happy coding.
